.github/workflows/main.yml

```
@@ -0,0 +1,23 @@
+name: Build and Check Format
+
+on:
+  pull_request:
+    branches: [ main ]
+
```

```suggestion
  push:
    branches: 
      - main

```

mainにプッシュした際もトリガーに含めたいかも。
例えばマージする際にrebaseした場合とかfast-forwardされた場合の安定性が保証できないかもと思った